backend/fms_core/viewsets/experiment_run.py

```diff
@@ -46,7 +47,21 @@ def list_export(self, _request):
 
     @action(detail=True, methods=["patch"])
     def launch_run_processing(self, _request, pk=None):
-        experiment_run, errors, warnings = launch_experiment_run(pk)
+        '''
+        Generates a run info file for an experiment, which triggers run processing
+        and sets the experiment's run processing launch time to the current date.
+
+        Args:
+            The experiment ID.
+
+        Returns:
+            On success:
+            {'ok': True}
+
+            On error:
+            {'ok': False, message: <error message>}
+        '''
+        experiment_run, errors, warnings = start_experiment_run_processing(pk)
```

Comment:
The 2 services created in experiment_run are not really part of experiment_run. I would move the code from those 2 services to the viewsets and get rid of those 2 experiment_run services. They are just a redirection of call to experiment_run_info. As a more general comment, a function need to accomplish a function. There is too many layers of paper wrapper here.

Reply:
I can change that, but I don't really agree. To me, the responsibility of a viewset should be to provide an end point and handle the request parameters and the response. I prefer a cleaner separation of concerns - http mechanics in the viewset, business logic in the service. Otherwise you can end up with business logic locked up in a viewset, where the code cannot be reused, or which gets copy-pasted later.

Reply:
you are right but that would mean we have already a lot of refactoring to do to reach that point for the other viewsets. The idea is not that i want that code in viewset or any specific place it's more that it takes 3 files changes to get to any actual code ... The distinction between viewset and the services i agree with. But i don't see why you have all those cascading function where you basically have a single functionality. The way i see it you treat experiment_run_info as an object class without making one of it. Maybe you can define it more officially and we can make a home for it (outside of services, maybe a new reporting package). That way you could use the constructor to build the run_info then you can have .tojson, .tofile, .torunprocessingspool methods to call in the service.